cloud/vsphere/machineactuator.go

```diff
@@ -739,17 +783,57 @@ func getKubeadmToken() (string, error) {
 	return strings.TrimSpace(output), err
 }
 
+func (vc *VsphereClient) getMasterForCluster(cluster *clusterv1.Cluster) (*clusterv1.Machine, error) {
+	machines, err := vc.lister.Machines().Lister().Machines(cluster.Namespace).List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	for _, machine := range machines {
+		//glog.Infof("%v", machine)
+		if util.IsMaster(machine) {
+			// Return the first master for now. Need to handle the multi-master case better
+			glog.Infof("Found the master VM %s", machine.Name)
+			return machine, nil
+		}
+	}
+	return nil, fmt.Errorf("No master node found for the cluster %s", cluster.Name)
+}
+
+func createTempFile(contents string) (string, error) {
+	f, err := ioutil.TempFile("", "")
+	if err != nil {
+		return "", err
+	}
+	defer ifErrRemove(&err, f.Name())
+	if err = f.Close(); err != nil {
+		return "", err
+	}
+	err = ioutil.WriteFile(f.Name(), []byte(contents), 0644)
+	if err != nil {
+		return "", err
+	}
+	return f.Name(), nil
+}
+
+func ifErrRemove(pErr *error, path string) {
```

Comment:
you probably should make this an inline function in the above function since it's very specific to the temp file you're creating.

defer func() {
}()